doc/release/upcoming_changes/22707.improvement.rst

```diff
@@ -0,0 +1,9 @@
+New ``DTypePromotionError``
+---------------------------
+NumPy now has a new ``DTypePromotionError`` which is used when two
+dtypes cannot be promoted to a common one, for example raised by::
+
+    np.result_type("M8[s]", np.complex128)
+
+Errors that occur while trying to find the correct promotion may raise a
+different error.
```

Comment:
This sentence is incomplete. Either remove it, or give more detail: what other error class can be emitted?

Reply:
Yeah, this is mainly the weirder datetime unit conversions that fail.  But maybe that is all obscure enough that removing the sentence is best, since I don't want to give the example.
(Will remove it)